### DISCUSSION TITLE
关于脉冲神经网络数据集的问题

### DISCUSSION THREAD

**Discussion by Spice-monkey** [fangwei123456/SpikingFlow#392] [Q&A]
您好，我看到您提供了一些数据集，在其他资料上我也看到了一些关于脉冲编码的方式，大致有概率编码，延时编码，增量编码。我的个人理解是增量编码是不是就是事件驱动的编码方式。具体来说，我理解的手写数字集的神经形态版本就是事件驱动编码的吧，但是感觉这样是不是很麻烦，个人感觉对于一个静态图像如果想要用这种方式编码是不是不太适合？或者说如果我想创建一些静态图像的数据集，应该怎么做？

**Comment by fangwei123456** [comment]
> 我理解的手写数字集的神经形态版本就是事件驱动编码的吧

是用神经形态相机拍摄静态图片得到的，具体请参考原论文。

> 或者说如果我想创建一些静态图像的数据集，应该怎么做？

参考pytorch的教程

**Reply by Spice-monkey** [reply]
您好，我参考的论文上面似乎是搭建了一个平台，固定图像，之后移动相机得到的

**Reply by Spice-monkey** [reply]
如果我想把一些静态图片变成脉冲神经网络可接受的数据集怎么做？我也必须利用神经形态相机吗

**Reply by fangwei123456** [reply]
如果想得到纯01数据，可以考虑用SJ框架内的编码器，例如泊松/延迟编码器等

**Reply by Spice-monkey** [reply]
您的意思是SJ框架能够直接将这静态图片通过泊松或延迟编码成脉冲序列对吗

**Reply by fangwei123456** [reply]
是的

**Reply by Spice-monkey** [reply]
如果想利用增量编码，也就是对比t时刻和t+1时刻的数据变化是否大于阈值从而产生脉冲这种形式，有什么办法吗

**Reply by Spice-monkey** [reply]
我感觉是必须从硬件层面入手？只靠算法处理是不是完成不了类似的转换

**Reply by fangwei123456** [reply]
用一层IF神经元，每个时刻输出后都重置神经元即可

**Reply by Spice-monkey** [reply]
这种数据集的内容是记录每个时间步对应的每个像素的坐标以及是否发出了脉冲是吗？x,y,t,p这种格式？

**Reply by fangwei123456** [reply]
不能，需要自行转换

**Reply by Spice-monkey** [reply]
请问有什么转换的方法或者论文参考吗？

**Reply by fangwei123456** [reply]
无，需要自行设计
